libsql-wal/src/bins/shell/main.rs

```diff
@@ -204,33 +232,30 @@ async fn handle_builtin<S>(
     false
 }
 
-struct Env<S, B: Backend> {
-    registry: WalRegistry<StdIO, S>,
-    backend: Arc<B>,
-}
-
-async fn setup_s3_registry(
-    db_path: &Path,
-    bucket_name: &str,
-    cluster_id: &str,
-    url: &str,
-    region_id: &str,
-    access_key_id: &str,
-    secret_access_key: &str,
+async fn setup_s3_storage(
+    cli: &Cli,
     join_set: &mut JoinSet<()>,
-) -> Env<AsyncStorage<S3Config, SealedSegment<std::fs::File>>, S3Backend<StdIO>> {
-    let cred = Credentials::new(access_key_id, secret_access_key, None, None, "");
+) -> AsyncStorage<S3Backend<StdIO>, SealedSegment<std::fs::File>> {
+    let cred = Credentials::new(
+        cli.s3_args.s3_access_key_id.as_ref().unwrap(),
+        cli.s3_args.s3_access_key.as_ref().unwrap(),
+        None,
+        None,
+        "",
+    );
     let config = aws_config::SdkConfig::builder()
         .behavior_version(BehaviorVersion::latest())
-        .region(Region::new(region_id.to_string()))
+        .region(Region::new(
+            cli.s3_args.s3_region_id.as_ref().unwrap().to_string(),
+        ))
         .credentials_provider(SharedCredentialsProvider::new(cred))
-        .endpoint_url(url)
+        .endpoint_url(cli.s3_args.s3_url.as_ref().unwrap())
         .build();
     let backend = Arc::new(
         S3Backend::from_sdk_config(
             config.clone(),
-            bucket_name.to_string(),
-            cluster_id.to_string(),
+            cli.s3_args.s3_bucket.as_ref().unwrap().to_string(),
+            cli.s3_args.cluster_id.as_ref().unwrap().to_string(),
         )
```

Comment:
I think this should use the creds resolver that is built in so that it pulls aws creds from your `~/.aws/` dir or env vars instead of doing custom credential resolution.

Reply:
do you know how to do that? can you submit a PR?